test/e2e/gc_e2e_test.go

```diff
@@ -34,159 +33,268 @@ var _ = Describe("Garbage collector", func() {
-		Expect(err).NotTo(HaveOccurred())
-		defer func() {
-			IgnoreError(kubeClient.ApiextensionsInterface().ApiextensionsV1().CustomResourceDefinitions().Delete(context.TODO(), crd.GetName(), metav1.DeleteOptions{}))
-		}()
-
-		cr, err := kubeClient.CreateClusterRole(&rbacv1.ClusterRole{
-			ObjectMeta: metav1.ObjectMeta{
-				GenerateName:    "clusterrole-",
-				OwnerReferences: []metav1.OwnerReference{ownerutil.NonBlockingOwner(crd)},
-			},
+				}, metav1.CreateOptions{})
+				Expect(err).NotTo(HaveOccurred())
+
+				defer func() {
+					IgnoreError(kubeClient.ApiextensionsInterface().ApiextensionsV1().CustomResourceDefinitions().Delete(context.TODO(), crd.GetName(), metav1.DeleteOptions{}))
+				}()
+
+				// Create a ClusterRole for the crd
+				cr, err := kubeClient.CreateClusterRole(&rbacv1.ClusterRole{
+					ObjectMeta: metav1.ObjectMeta{
+						GenerateName:    "clusterrole-",
+						OwnerReferences: []metav1.OwnerReference{ownerutil.NonBlockingOwner(crd)},
+					},
+				})
+				Expect(err).NotTo(HaveOccurred())
+
+				defer func() {
+					IgnoreError(kubeClient.DeleteClusterRole(cr.GetName(), &metav1.DeleteOptions{}))
+				}()
+
+				Expect(kubeClient.ApiextensionsInterface().ApiextensionsV1().CustomResourceDefinitions().Delete(context.TODO(), crd.GetName(), metav1.DeleteOptions{})).To(Succeed())
+				Eventually(func() bool {
+					_, err := kubeClient.GetClusterRole(cr.GetName())
+					return k8serrors.IsNotFound(err)
+				}).Should(BeTrue(), "get cluster role should eventually return \"not found\"")
+			})
+
 		})
-		Expect(err).NotTo(HaveOccurred())
-		defer func() {
-			IgnoreError(kubeClient.DeleteClusterRole(cr.GetName(), &metav1.DeleteOptions{}))
-		}()
-
-		Expect(kubeClient.ApiextensionsInterface().ApiextensionsV1().CustomResourceDefinitions().Delete(context.TODO(), crd.GetName(), metav1.DeleteOptions{})).To(Succeed())
-		Eventually(func() bool {
-			_, err := kubeClient.GetClusterRole(cr.GetName())
-			return k8serrors.IsNotFound(err)
-		}).Should(BeTrue(), "get cluster role should eventually return \"not found\"")
 	})
 
-	It("should delete a ClusterRole owned by an APIService when the owner is deleted", func() {
-		group := rand.String(16)
-		as, err := kubeClient.CreateAPIService(&apiregistrationv1.APIService{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: fmt.Sprintf("v1.%s", group),
-			},
-			Spec: apiregistrationv1.APIServiceSpec{
-				Group:                group,
-				Version:              "v1",
-				GroupPriorityMinimum: 1,
-				VersionPriority:      1,
-			},
-		})
-		Expect(err).NotTo(HaveOccurred())
-		defer func() {
-			IgnoreError(kubeClient.DeleteAPIService(as.GetName(), &metav1.DeleteOptions{}))
-		}()
-
-		cr, err := kubeClient.CreateClusterRole(&rbacv1.ClusterRole{
-			ObjectMeta: metav1.ObjectMeta{
-				GenerateName:    "clusterrole-",
-				OwnerReferences: []metav1.OwnerReference{ownerutil.NonBlockingOwner(as)},
-			},
+	Context("Given a ClusterRole owned by a APIService", func() {
```

Comment:
Since the APIService spec is essentially the same as the CRD one, my other comments apply equally here.